pkg/controllers/machine_controller.go

```diff
@@ -656,10 +659,17 @@ func (r *MachineReconciler) setDomainResources(machine *api.Machine, domain *lib
 		Unit:  "Byte",
 	}
 
+	if !r.disableHugepages {
+		domain.MemoryBacking = &libvirtxml.DomainMemoryBacking{
+			MemoryHugePages: &libvirtxml.DomainMemoryHugepages{},
+		}
+	}
+
```

Comment:
I would propose to keep only this for now until we handle the resources correctly. The other changes will break contracts and we announce incorrect resources.

Reply:
it can create wrong behavior because same machines will use hugepages other not. So you can have performance differences between same type of machine. What can be confusing for users. And i am not sure, if we can create machine without free hugepages, Adree mentioned something it is possible, but i didn't test it. But you will have problem with different performance.

Reply:
we will have problem with different performance, i minimized problem with usage of total count of hugepages for calculate quantity but problem still exists.